src/main/kotlin/org/jetbrains/bio/big/Bed.kt

```
@@ -357,4 +357,26 @@ data class ExtendedBedEntry(
         }
         return rest
     }
+
+    operator fun get(i: Int, fieldsNumber: Int, extraFieldsNumber: Int): Any? {
+        return when {
+            i >= fieldsNumber + extraFieldsNumber -> null
+            i >= fieldsNumber -> extraFields?.get(i - fieldsNumber)
+            else -> when (i) {
+                0 -> start
+                1 -> end
+                2 -> chrom
+                3 -> name
+                4 -> score
+                5 -> strand
+                6 -> thickStart
+                7 -> thickEnd
+                8 -> itemRgb
+                9 -> blockCount
+                10 -> blockSizes
+                11 -> blockStarts
+                else -> null // should never happen
```

Please throw an exception here with a useful message if it happens. If smb refactor the above code it's better to see exception instead of unexpected NPE

---

Now that I think of it, the comment is misleading. This branch will happen e.g. if `i` is negative, which is exactly an expected result for this method. It clearly says in javadoc that the out-of-bounds value is `null`. I'd say that since the return type is nullable, it's the receiver's responsibility to check for it.
If you think that we should throw instead of returning null, I'd be glad to hear why.

---

After a Slack discussion, we've decided that `null` is okay here.